experimental/parallel_compose.cpp

```diff
@@ -910,6 +910,12 @@ Graph compose(const Graph& first, const Graph& second) {
       }
     }
   }
+
+  // Shift offset values back down after adding arcs to newGraphDP
+  for (int i = newGraphDP.outArcOffset.size() - 1; i >= 0; --i) {
+    newGraphDP.outArcOffset[i] = i == 0 ? 0 : newGraphDP.outArcOffset[i - 1];
+    newGraphDP.inArcOffset[i] = i == 0 ? 0 : newGraphDP.inArcOffset[i - 1];
+  }
```

Comment:
PS I think we need something like this after the increments at L251/251, but lmk.